pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -285,6 +290,42 @@ func (s *Server) initPublicKey() error {
 	return nil
 }
 
+// loadRemoteCACerts mounts an existing cacerts Secret if the files aren't mounted locally.
+// By default, a cacerts Secret would be mounted during pod startup due to the
+// Istiod Deployment configuration. But with external Istiod, we want to be
+// able to load cacerts from a remote cluster instead.
+func (s *Server) loadRemoteCACerts(caOpts *CAOptions, dir string) error {
+	if s.kubeClient == nil {
+		return nil
+	}
+
+	signingKeyFile := path.Join(dir, "ca-key.pem")
+	if _, err := os.Stat(signingKeyFile); !os.IsNotExist(err) {
+		return fmt.Errorf("signing key file %s already exists", signingKeyFile)
+	}
+
+	secret, err := s.kubeClient.Kube().CoreV1().Secrets(caOpts.Namespace).Get(
+		context.TODO(), "cacerts", metav1.GetOptions{})
```

Comment:
What is the diff with `CASecret = "istio-ca-secret"`, it really makes me confusing

Reply:
Unfortunately I think either `cacerts` or `istio-ca-secret` can present remotely. We need to merge them and deprecate one. At the moment, I think you do need to query `istio-ca-secret` if the other one doesn't exist.

Reply:
I'm not totally sure of the difference... This code is only meant to simulate the mounting of `cacerts` though; the existing code for handling `istio-ca-secret` should still work when `cacerts` doesn't exist, as in the non-external-istiod case